interview-questions.md

```diff
@@ -8,42 +8,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: The principles of Object oriented programming are:  
 
-1. What is the difference between a Float and an Integer in Ruby?
+- Encapsulation, where you contain logic inside objects.
+- Abstraction, which allow only pertinent code to be shown while hiding unnecessary code.
+- Inheritance, which I meantioned before in where a parent class creates a blueprint for child classes.
+- Polymorphism, where child classes, while inheriting properties of the parent, can also other properties required for specialization.
 
-Your answer:
+2. What is the difference between a Float and an Integer in Ruby?
 
-Researched answer:
+Your answer: The difference between a float and an integer in Ruby is that a float requires a higher degree of precision, so a float variable requires more memory allocation than an integer.
 
-1. Ruby has an implicit return. What does that mean?
+Researched answer: in my researched answer I found the same thing in that a float provides a higher degree of preceision but an integer requires less memory from the program so it is faster to process, so it is worthwile using an integer when a high degree of accuracy is not required. 
 
-Your answer:
+3. Ruby has an implicit return. What does that mean?
 
-Researched answer:
+Your answer: In Ruby an implicit return means that the last line of a block is always assumed to be the return statement by the program, so it is not needed to expressly write "return" in order to receive the last line of the block as a return. 
 
-1. What is a block in Ruby?
+Researched answer: In my research I found the same in that the implicit return in Ruby will just return the last line of a block if an explicit return is not given. This is helpful but can also cause issues if one forgets about the implicit return and ends up getting an un-expected return from a block. 
 
-Your answer:
+4. What is a block in Ruby?
 
-Researched answer:
+Your answer: A block in Ruby is a "named" chunk of code that is contained in curly bracers { block } and can be invoked from a function. 
 
-1. How do you extract a value in a Ruby hash?
+Researched answer: Blocks are anonymous pieces of code that can take input from arguments and return a value. 
 
-Your answer:
+5. How do you extract a value in a Ruby hash?
 
-Researched answer:
+Your answer: I honestly do not remember at all how to manipulate a hash in Ruby. But please look below for my researched answer!
+
+Researched answer: In ruby a hash is collection of key:value pairs similar to dictionaries. You can extract both the keys and values by using a do - end statement as follows
+
+test_hash.each do |key, value|
+    puts " Here's the key #{key}, and here is the value #{value} "
+end
+
+alternatively if you only wanted the value you can use test_hash.[key] to get only that key's specific value
 
```

Comment:
Great example here! Perfectly fine to not remember in an interview, your answer to deal with that was great!